### ISSUE TITLE
command-line client fails in 5.0-RC8 when TERM is "linux"

### ISSUE THREAD

**Issue by craigberry** [eXist-db/exist#2951]
For a terminal or emulator that reports its type as "linux", `bin/client.sh --no-gui` with 5.0-RC8 on RHEL Linux crashes as below.  If the terminal type is "vt100" it all works fine.  It sounds very similar to this:

<https://github.com/jline/jline3/issues/408>

So maybe just upgrading to current jline would take care of it.

```
# echo $TERM
linux
# bin/client.sh --no-gui
eXist version 5.0.0-RC8 (b7a9e898163b8362e092ebfa5536c29032d2a93f), Copyright (C) 2001-2019 The eXist-db Project
eXist-db comes with ABSOLUTELY NO WARRANTY.
This is free software, and you are welcome to redistribute it
under certain conditions; for details read the license file.

java.lang.reflect.InvocationTargetException
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.exist.start.Main.invokeMain(Main.java:118)
	at org.exist.start.Main.runEx(Main.java:248)
	at org.exist.start.Main.run(Main.java:123)
	at org.exist.start.Main.main(Main.java:60)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.codehaus.mojo.appassembler.booter.AppassemblerBooter.executeMain(AppassemblerBooter.java:257)
	at org.codehaus.mojo.appassembler.booter.AppassemblerBooter.main(AppassemblerBooter.java:77)
Caused by: java.io.IOError: java.lang.UnsupportedOperationException
	at org.jline.utils.Curses.tputs(Curses.java:62)
	at org.jline.utils.Curses.tputs(Curses.java:45)
	at org.jline.keymap.KeyMap.key(KeyMap.java:243)
	at org.jline.reader.impl.LineReaderImpl.key(LineReaderImpl.java:5784)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at org.jline.reader.impl.LineReaderImpl.bindKeys(LineReaderImpl.java:5792)
	at org.jline.reader.impl.LineReaderImpl.emacs(LineReaderImpl.java:5503)
	at org.jline.reader.impl.LineReaderImpl.defaultKeyMaps(LineReaderImpl.java:5479)
	at org.jline.reader.impl.LineReaderImpl.<init>(LineReaderImpl.java:277)
	at org.jline.reader.LineReaderBuilder.build(LineReaderBuilder.java:115)
	at org.exist.client.InteractiveClient.run(InteractiveClient.java:2223)
	at org.exist.client.InteractiveClient.main(InteractiveClient.java:243)
	... 14 more
Caused by: java.lang.UnsupportedOperationException
	at org.jline.utils.Curses.doTputs(Curses.java:78)
	at org.jline.utils.Curses.tputs(Curses.java:60)
	... 33 more
```

**Comment by dizzzz** [comment]
v3.12.1 is available.

**Comment by craigberry** [comment]
I can confirm this is fixed as of `eXist-5.0.0-RC8-56-g29293ce97`.
